### PR TITLE
[PUBDEV-6821] Expose leader node via REST API

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

**Comment by Pscheidl** [comment]
Works great. I propose adding the following sanity test (would be great for the whole endpoint).
```patch
Index: h2o-core/src/test/java/water/api/CloudHandlerTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- h2o-core/src/test/java/water/api/CloudHandlerTest.java	(date 1566814336492)
+++ h2o-core/src/test/java/water/api/CloudHandlerTest.java	(date 1566814336492)
@@ -0,0 +1,24 @@
+package water.api;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.TestUtil;
+import water.api.schemas3.CloudV3;
+
+import static org.junit.Assert.assertNotNull;
+
+public class CloudHandlerTest extends TestUtil {
+  
+  @BeforeClass
+  public static void beforeClass(){
+    TestUtil.stall_till_cloudsize(1);
+  }
+  
+  @Test
+  public void status(){
+    final CloudHandler cloudHandler = new CloudHandler();
+    final CloudV3 status = cloudHandler.status(3, new CloudV3());
+    assertNotNull(status);
+    assertNotNull(status.leader);
+  }
+}

```

**Comment by jakubhava** [comment]
@Pscheidl great suggestion, done

File: h2o-core/src/main/java/water/api/CloudHandler.java [hunk 1]
```diff
@@ -41,6 +41,11 @@ public CloudV3 status(int version, CloudV3 cloud) {
     H2ONode[] members = H2O.CLOUD.members();
     cloud.bad_nodes = 0;
     cloud.cloud_healthy = true;
+    
+    H2ONode leader = H2O.CLOUD.leader();
+    if (leader != null) {
+      cloud.leader = new CloudV3.NodeV3(leader, cloud.skip_ticks);
```
[the remark this answers is not in the document]
Comment (jakubhava) [reply h2o-core/src/main/java/water/api/CloudHandler.java]: 👍 I like the leader_idx approach, thanks @michalkurka

**Comment by jakubhava** [comment]
@Pscheidl @michalkurka please have a look, applied the changes. Thanks!

**Comment by Pscheidl** [comment]
Thanks @jakubhava . I can only approve. :heavy_check_mark:

**Pull request merged**